### DISCUSSION TITLE
v-virtual-scroll does not update when items changes

### DISCUSSION THREAD

**Discussion by cadilhac** [vuetifyjs/vuetify#18545] [Q&A]
I would like to know if it's normal than v-virtual-scroll does not update when its items property changes (in my case, items are reordered). Items is linked to a component prop. If yes, how do I do to update the scroller when the items prop changes?

```
<v-virtual-scroll :items="games">
    <template v-slot:default="{ item }">
        <PersonalGameCard :game="item">
        </PersonalGameCard>
    </template>
</v-virtual-scroll>

const props = defineProps(['games'])
```

Also is there a notion of key? No sign of this in the doc. Intuitively, I would have added :key="item.id" to my PersonalGameCard element...

Finally, note that if I replace the virtual scroller by a basic v-for, then it works perfectly.

Thanks

**Comment by yuwu9145** [comment]
Reproduce link plz

**Reply by cadilhac** [reply]
Unfortunately I don't succeed writing a repro. In [my attempt](https://play.vuetifyjs.com/#eNqNVMFu2zAM/RVOwJAUiO2uAbbCTYoO287bYbe6B9emG22yJEiy0yLIv4+S7MxJgCJAkIQUyfeeKPJxx6ypsq9ap32HLGcrh60WpcP7QgKs+qTUOvwNRqWkK7lEM7iC0265qzYgymcU64L96FEWDPqkVXVwYHDcr7Ix9H9ypVoN+UvZoqXAhguHBuvvpSupQjbiUuIxsPdEXqtswpdMWxmuHVh0XaTNW62Mgx0YbBbg8TqHNeyhMaqFGYmeTcICn3iSZt/I8LdCAT6EKFgHXgysfbW5Mx1eTc9q4j2cPUbqu1Epr3P4tBgtSYJzmP2UOBt8++H3KOHmNOH3Vr2bsDxL2Bg8wXgiyiPh6YUT8fF25vMrWN+PpXkDc6867UtBgkme64wMYqMrjWXmrz7pNeU1fIQbWK/XcB2wAFBYvDTvwyFvT9+rLHaUeskWLHYpaUud/rFK0nMNHIvhwBYsH1kXjDrn7YJtnNM2z7KqlpRGb5L3JpXoMqnb7IHCMtNJx1tMatU+LNNl+iWruXVTd4q2TZ6N2lo0VKRgw50GGMebtwughsiAcLOMEIMvoeGxHuOsdkaAPZrEoKypVeZSSSdpU1knR2fSwt0Xkj505X47jINwvh5q3oN1bwJpeqk2nbzl8CxU9fcONshfNi6H2+tr/XoHijAbobYJRZSdU7QRDmul58Z1pUio10oIyDmh+IUQFgNtAm8nsR55P9+Sa0Qdy+bAJXe8FIe6VHkkS8vICuXyGpuyE77GLtQkmYzeeTRSPzOw39N2ORIZ180xxWELkfxLdlAcNm2UtjRlRIIW2S9vzR9nQeLMD+X0qe8XLDSMGhBfC3v6B6HX2M4=), it works well, which frustrates me.
I also realized that in my own app, the scroller updates the new items on screen only if I resize it, otherwise it does not.

**Reply by cadilhac** [reply]
ok, I got it and it has to do with Vue, not Vuetify. I changed the [repro to make it fail](https://play.vuetifyjs.com/#eNqVVF1vmzAU/StXfgmVArSNtFU0zTpt75u0vZU+OGAab8a2bENWRfz3Xdsw0qSaWikK+H6dey73+OFArKnyz1pnfcdIQdaOtVpQxzalBFj3KdU6vIZDpaSjXDIzmtBYqVZD8URbZu9KUlNHSwL5lJKf5nhLLLnOj6DwaCvDtQPLXBcReauVcXAAwxoYoDGqhQV2uThyBvToyfIvePA0MMCHILB14DuCO3iIDR2mvnldwNVyOklsv4DFN8kWo20Yny8Srk8Tfu7VfxNWZwk7w04wHv0oInmkTZYkUktbqrNfVkn8KKFmOTpsSYoJpSRI159LsnNO2yLPq1piWs0E700mmculbvN7DMtNJx1vWVqr9n6VrbKPec2tOzZnzLbp1qi9ZQaLlGRsNMA43jy/AWqMDAjXqwgx2lJBt9ZjnNXOEbBnJjVM1sww81ZKJ2nHtE5cZ9Q8+lBK/OHIvQam7TkXQc17sO5ZML/g3KLnuYCtUNXvW9gx/rRzBdxcXuo/t6AQsxFqn2IE7ZwqySweu+eu2gEOgQms9AO/JUqlT1uF1NBgg2HjFRJD59SeG9dRkeKaKCGmnSo49ulV13DhkGT9Najv3wKiN439YcyHm9kzkZm6LYBL7jgVU8gs72kS2KcVyhU1a2gnfMFDAMAZkg0c4iHzWw7DgCp/McEo+5ckxtsAZ/uuu8ALvnOsXr5yK8yi10Zpi6rHbvHm+e5PycMiXFKLx4s5zo8cw7BU4kzH0DP7joeKMRNwklzAHTKOvHgDiS+S9VRgPlZynZERPwt4mXcnCV3CNiQmNOM1bGDrH5/gCgpIry5CTwBMWPbmGuvXagz4f3ydDEsSRIFLHhVJHv8CxcrsUw==)

It has to do with the initial data that is not reactive and my computed value that uses inplace sort. In that case, I should make it reactive or make a copy of the array before calling sort.
